src/gentropy/dataset/study_index.py

```diff
@@ -390,3 +392,40 @@ def validate_target(self: StudyIndex, target_index: GeneIndex) -> StudyIndex:
         )
 
         return StudyIndex(_df=validated_df, _schema=StudyIndex.get_schema())
+
+    def validate_dataset(
```

Comment:
Can this method be moved into the `Dataset` class? It looks very similar to `StudyLocus.validate_dataset`.